src/biogeochem/DUSTMod.F90

```diff
@@ -924,4 +1219,67 @@ subroutine InitDustVars(this, bounds)
 
   end subroutine InitDustVars
 
+  !==============================================================================
+  subroutine CalcDragPartition(this, bounds, prigent_roughness_stream )
```

Comment:
Actually, maybewe can simply pass the `prigent_roughness_stream%prigent_rghn` to DUSTMod.F90 and calculate the drag partition effect in the DustEmission subroutine, instead of inside InitCold. Unless other dust emission scheme wants to use this subroutine. Just a thought, totally optional.

Reply:
The advantage of leaving it inside InitCold is then it's only done once at initialization. Since, it doesn't need to be updated it would be good to make sure we only do it once. If it ever does need to be updated, we'd want to do it every time DustEmission is called.